server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -4043,6 +4044,8 @@ private Pair<List<Long>, Integer> searchForServiceOfferingIdsAndCount(ListServic
         SearchCriteria<ServiceOfferingVO> sc = serviceOfferingSearch.create();
         if (state != null) {
             sc.setParameters("state", state);
+        } else {
+            sc.setParameters("state", Arrays.asList(ServiceOffering.State.Active, ServiceOffering.State.Inactive));
```

Comment:
Same here, if you do not set this parameter, both active and inactive offerings will be queried.